src/background.js

```diff
@@ -266,6 +268,26 @@ async function maybeReopenTab (url, tab, request) {
   return {cancel: true};
 }
 
+const rootDomainCache = {};
+
+function getRootDomain(url) {
+  if (url in rootDomainCache) {
+    // After storing 128 entries, it will delete the oldest each time.
+    if (Object.keys(rootDomainCache).length > 128) {
+      delete rootDomainCache[(Object.keys(rootDomainCache)[0])];
+    }
+    return rootDomainCache[url];
+  }
```

Comment:
If `rootDomainCache[url]` is the `0`th key in the object, it will be deleted and then this will return `null`? So, store the `rootDomainCache[url]` value in a return variable before ejecting it from the memo:

```suggestion
    // After storing 128 entries, it will delete the oldest each time.
    const returnValue = rootDomainCache[url];
    if (Object.keys(rootDomainCache).length > 128) {
      delete rootDomainCache[(Object.keys(rootDomainCache)[0])];
    }
    return returnValue;
  }
```